src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt

```diff
@@ -86,33 +86,32 @@ fun processItemDeclarations(
         }
     }
 
-    if (Namespace.Types in ns) {
-        if (scope is RsFile && scope.isCrateRoot && withPrivateImports) {
-            // Rust injects implicit `extern crate std` in every crate root module unless it is
-            // a `#![no_std]` crate, in which case `extern crate core` is injected. However, if
-            // there is a (unstable?) `#![no_core]` attribute, nothing is injected.
-            //
-            // https://doc.rust-lang.org/book/using-rust-without-the-standard-library.html
-            // The stdlib lib itself is `#![no_std]`, and the core is `#![no_core]`
-            when (scope.attributes) {
-                RsFile.Attributes.NONE ->
-                    if (processor.lazy(STD) { scope.findDependencyCrateRoot(STD) }) return true
-
-                RsFile.Attributes.NO_STD ->
-                    if (processor.lazy(CORE) { scope.findDependencyCrateRoot(CORE) }) return true
-
-                RsFile.Attributes.NO_CORE -> Unit
-            }
-        }
-    }
-
     for (speck in itemImports) {
         check(speck.useGroup == null)
         val path = speck.path ?: continue
         val name = speck.nameInScope ?: continue
         if (processMultiResolveWithNs(name, ns, path.reference, processor)) return true
     }
 
+    if (Namespace.Types in ns && scope is RsMod && withPrivateImports) {
+        // Rust injects implicit `extern crate std` in every module unless it is
+        // a `#![no_std]` crate, in which case `extern crate core` is injected. However, if
+        // there is a (unstable?) `#![no_core]` attribute, nothing is injected.
+        //
+        // https://doc.rust-lang.org/book/using-rust-without-the-standard-library.html
+        // The stdlib lib itself is `#![no_std]`, and the core is `#![no_core]`
+        if (!scope.isCrateRoot && processor(ScopeEvent.IMPLICIT_CRATES)) return true
```

Comment:
Hm, could you remind why we handle stdlib implicit crates here but all other code related to extern crates is located in [NameResolution](https://github.com/intellij-rust/intellij-rust/blob/31e7b1303e79637a9e95b36f7b04aa82c13f2582/src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt#L470-L484)?
Maybe it possible to handle them in one place?

Reply:
Yeah, we should move them here. I don't sure I should do it in a separate PR or not